### PR TITLE
Add end to end test for filtering

### PR DESCRIPTION
**Pull request by peterhudec**

### CODE REVIEW

File: cypress/e2e_tests/specs/filter.test.js [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+    [pk]: `${first_name} ${last_name}`,
+  }),
+  {},
+)
+
+// The result list items only show a subset of the enquiry data and we need
+// to make assertions about the information not shown. The constants
+// ENQUIRIES and ENQUIRERS constants are maps of primary keys to enquiries and
+// enquirers fixtures respectively, which we are using to make assertions.
+const fixtureReducerc = (a, {pk, fields}) => ({ ...a, [pk]: fields })
```
Comment (rafenden) [inline cypress/e2e_tests/specs/filter.test.js:16]: Typo at the end.
```suggestion
const fixtureReducer = (a, {pk, fields}) => ({ ...a, [pk]: fields })
```

File: cypress/e2e_tests/specs/filter.test.js [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+  added: 'Added to Data Hub',
+  nonFdi: 'Non-FDI',
+  nonResponsive: 'Non-responsive',
+  awaiting: 'Awaiting response from Investor',
+  postProgressing: 'Post progressing',
+  sent: 'Sent to Post',
+  engaged: 'Engaged in dialogue',
+}
+
+const filterLabels = filters =>
```
Comment (ian-leggett) [inline cypress/e2e_tests/specs/filter.test.js:35]: Is it worth moving this to `commands.js`

File: cypress/e2e_tests/specs/filter.test.js [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+  Object.entries(filters)
+    .map(([k, v]) =>
+      typeof v === 'string'
+        ? `${k}: ${v}`
+        : k
+     )
+    .join(', ')
+
+const getInputByLabel = (label) =>
```
Comment (ian-leggett) [inline cypress/e2e_tests/specs/filter.test.js:44]: Is it worth moving this to `commands.js`

File: cypress/e2e_tests/specs/filter.test.js [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+  cy.contains('label', label)
+    .parent()
+    .find('input')
+
+const setFilters = filters =>
```
Comment (ian-leggett) [inline cypress/e2e_tests/specs/filter.test.js:49]: Is it worth moving this to `commands.js`

File: cypress/e2e_tests/specs/filter.test.js [hunk 5]
```diff
@@ -0,0 +1,558 @@
...
+        // Date input doesn't allow empty string to be typed in
+        ? input.clear()
+        : input.type(val)
+      : val
+        ? input.check()
+        : input.uncheck()
+
+  })
+
+const assertResults = (assert, expectedTotal) => {
```
Comment (ian-leggett) [inline cypress/e2e_tests/specs/filter.test.js:63]: Is it worth moving this to `assertions.js`

File: cypress/e2e_tests/specs/filter.test.js [hunk 6]
```diff
@@ -0,0 +1,558 @@
...
+    .contains(`${expectedTotal} enquiries`)
+
+  expectedTotal && cy.get('article')
+    .find('ol li')
+    .as('items')
+    .should('have.length', expectedTotal > 10 ? 10 : expectedTotal)
+    .each(assert)
+}
+
+const testFilters = ({ filters, assertItem, expectedTotal }) => {
```
Comment (ian-leggett) [inline cypress/e2e_tests/specs/filter.test.js:74]: Is it worth moving this to `assertions.js`

Comment (peterhudec) [reply cypress/e2e_tests/specs/filter.test.js:74]: The thing is that this function and all the other ones you mention in the above comments only make sense in this spec file and I would only move generic stuff to `./support`. A possible candidate would be the `getInputByLabel`, which could be very useful, but unfortunately, it is not trivial to make it work with all possible configuration of inputs and labels. The `getInputByLabel` here assumes a certain DOM structure of the filters so it's not really reusable.

**Pull request merged**